src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -77,20 +91,30 @@ internal static class GoToDefinitionHelpers
                     cancellationToken: cancellationToken);
             }
 
+            var navigableItems = preferredSourceLocations.Select(location => NavigableItemFactory.GetItemFromSymbolLocation(solution, symbol, location)).ToImmutableArray();
+            return TryGoToDefinition(navigableItems, title, project.Solution.Workspace, presenters, throwOnHiddenDefinition);
+        }
+
+        public static bool TryGoToDefinition(
+            ImmutableArray<INavigableItem> navigableItems,
+            string title,
+            Workspace workspace,
+            IEnumerable<Lazy<INavigableItemsPresenter>> presenters,
+            bool throwOnHiddenDefinition)
+        {
             // If we have a single location, then just navigate to it.
-            if (preferredSourceLocations.Length == 1)
+            if (navigableItems.Length == 1 && navigableItems[0].Document != null)
```

Comment:
When we do have navigable items with null documents?  The point of navigable items were that they were effectively a Document+Span pair.